### PR TITLE
feat: add x/nobleforwarding module for automated ica

### PR DESCRIPTION
**Pull request by jacobhjkim**
add x/nobleforwarding module for automated ica

spec on `x/forwarding` module is outlined here: https://hackmd.io/@noble/automated-ica

### CODE REVIEW

File: proto/settlus/nobleforwarding/tx.proto [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  // Register defines a rpc handler for MsgRegisterAccount.
+  rpc RegisterAccount(MsgRegisterAccount) returns (MsgRegisterAccountResponse);
+}
+
+// MsgRegisterAccount defines the payload for Msg/RegisterAccount.
+message MsgRegisterAccount {
+  option (cosmos.msg.v1.signer) = "sender";
+
+  string sender = 1;
+  string port = 2;
```
Comment (johnletey) [inline proto/settlus/nobleforwarding/tx.proto:19]: This isn't needed as the port will always be `transfer`.

File: proto/settlus/nobleforwarding/tx.proto [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  string channel_id = 3;
+  uint64 timeout_timestamp = 4;
```
Comment (johnletey) [inline proto/settlus/nobleforwarding/tx.proto:21]: To make it easier on integrators, this is also not necessarily needed. You can simply use the `transfertypes.DefaultRelativePacketTimeoutTimestamp` constant.

File: proto/settlus/nobleforwarding/packet.proto [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+    NoData noData = 1;
+    ForwardingAccountPacket forwardingAccountPacket = 2;
+  }
+}
+
+message NoData {}
+
+// ForwardingAccountPacket defines a struct for the noble forwarding account
+// packet payload
+message ForwardingAccountPacket { string registrant = 1; }
```
Comment (johnletey) [inline proto/settlus/nobleforwarding/packet.proto:17]:
```suggestion
message ForwardingAccountPacket { string recipient = 1; }
```

This has been changed to `recipient` 😄

**Review by johnletey — COMMENTED** [review]
Some of my initial thoughts @jacobhjkim 🙌🏻 

Great work so far on this!

**Comment by jacobhjkim** [comment]
closing this PR since we are going to use a different method for registering `forwardingAccount` on Noble.

**Pull request closed without merging**